app/interactors/schedule/save_datetime_interactor.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class Schedule::SaveDatetimeInteractor
+  include Interactor
+  delegate :params,
+           :edition,
+           :user,
+           :issues,
+           :datetime,
+           to: :context
+
+  def call
+    Edition.transaction do
+      find_and_lock_edition
+      check_for_issues
+      set_scheduled_publishing_datetime
+    end
+  end
+
+private
+
+  def find_and_lock_edition
+    context.edition = Edition.lock.find_current(document: params[:document])
+  end
+
+  def check_for_issues
+    unless edition.draft? || edition.submitted_for_review?
+      raise "Can't save a scheduling date unless edition is a draft or has been submitted for 2i"
+    end
+
+    checker = Requirements::ScheduledDatetimeChecker.new(params[:schedule])
+    pre_issues = checker.pre_submit_issues.issues
+
+    issues = Requirements::CheckerIssues.new(pre_issues + action_issues)
+
+    context.fail!(issues: issues) if issues.any?
+
+    context.datetime = checker.parsed_datetime
+  end
+
+  def set_scheduled_publishing_datetime
+    updater = Versioning::RevisionUpdater.new(edition.revision, user)
+    updater.assign(scheduled_publishing_datetime: context.datetime)
+
+    if updater.changed?
+      edition.assign_revision(updater.next_revision, user).save!
+      create_timeline_entry
+    end
```

Comment:
I'd suggest we use fail to abort if there aren't any changes, as per: https://github.com/alphagov/content-publisher/blob/93d067aea5b3f1b11c854b692efc38753a59dc8b/app/interactors/images/update_interactor.rb#L57

Then we can do timeline entry as a separate step

Reply:
Done 👍